File: src/editors/rectangle.ts
import * as sharp from "sharp"

export default {
    name: "Rectangle",
    description: "Draws a rectangle on the screen.",
    icon: "rectangle.png",
    expectsImageData: true,
    apply: async(partBuff: Buffer, rgb: Array<Number>): Promise<Buffer>  => {
        const metadata = await sharp(partBuff).metadata()
        return sharp({
            create: {
                width: metadata.width,
                height: metadata.height,
                background: { r: rgb[0], g: rgb[1], b: rgb[2], },
                channels: 4,
            },
        }).png().toBuffer()
    },
}

File: core/editors/rectangle.go
package editors

import (
	"github.com/magiccap/MagicCap/core/utils"
	"image"
)

func init() {
	Editors["rectangle"] = &Editor{
		Name:        "Rectangle",
		Description: "Draws a rectangle on the screen.",
		Icon:        utils.MustBytes(EditorAssets, "rectangle.png"),
		Apply: func(Region *image.RGBA, RGB [3]uint8) *image.RGBA {
			i := image.NewRGBA(Region.Rect)
			Y := 0
			Block := 0
			for Region.Bounds().Dy() != Y {
				X := 0
				for Region.Bounds().Dx() != X {
					i.Pix[Block] = RGB[0]
					i.Pix[Block+1] = RGB[1]
					i.Pix[Block+2] = RGB[2]
					i.Pix[Block+3] = 255
					Block += 4
					X++
				}
				Y++
			}
			return i
		},
	}
}
